qiskit-superstaq/qiskit_superstaq/superstaq_backend.py

```diff
@@ -136,3 +206,188 @@ def run(
         job = qss.SuperstaQJob(self, job_id)
 
         return job
+
+    def compile(
+        self,
+        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
+        num_equivalent_circuits: Optional[int] = None,
+        random_seed: Optional[int] = None,
+        atol: Optional[float] = None,
+        mirror_swaps: bool = True,
+        base_entangling_gate: str = "xx",
+        **kwargs: Any,
+    ) -> qss.compiler_output.CompilerOutput:
+        """Compiles the given circuit(s) to the backend's native gateset.
+
+        Args:
+            circuits: The qiskit QuantumCircuit(s) to compile.
+            num_equivalent_circuits: Number of logically equivalent random circuits to generate for
+                each input circuit.
+            random_seed: Optional seed for circuit randomizer.
+            atol: Tolerance to use for approximate gate synthesis (currently just for qutrit gates).
+            mirror_swaps: If mirror swaps should be used.
+            base_entangling_gate: The base entangling gate to use.
+            kwargs: Other desired compile options.
+
+        Returns:
+            A CompilerOutput object whose .circuit(s) attribute contains optimized compiled
+            circuit(s).
+
+        Raises:
+            ValueError: If `target` is not a valid AQT or IBMQ target.
+        """
+        _validate_qiskit_circuits(circuits)
+        if self.name().startswith("ibmq_"):
+            get_compiler_output = self.get_ibmq_compiler_output
+        elif self.name().startswith("aqt_"):
+            get_compiler_output = self.get_aqt_compiler_output
+        elif self.name().startswith("sandia_"):
+            get_compiler_output = self.get_qscout_compiler_output
+        elif self.name().startswith("cq_"):
+            get_compiler_output = self.get_cq_compiler_output
+        else:
+            raise ValueError(
+                f"{self.name()} is not a valid target (currently supports AQT, IBMQ, QSCOUT)."
+            )
+
+        options: Dict[str, Any] = {**kwargs}
+        if num_equivalent_circuits is not None:  # aqt eca compile
+            _validate_integer_param(num_equivalent_circuits)
+            options["num_eca_circuits"] = num_equivalent_circuits
+        if random_seed is not None:  # aqt eca compile
+            options["random_seed"] = random_seed
+        if atol is not None:  # aqt compile
+            options["atol"] = atol
+        if mirror_swaps is not None:  # qscout compile
+            options["mirror_swaps"] = mirror_swaps
+        if base_entangling_gate is not None:  # qscout compile
+            options["base_entangling_gate"] = base_entangling_gate
+        return get_compiler_output(circuits, options)
+
+    def get_aqt_compiler_output(
```

Comment:
it seems like these are doing more than just getting the compiler output?

i'm wondering if it makes sense to rename all the `get_xx_compiler_output` methods to `xx_compile`, and move all the target-specific argument handling and documentation to them, e.g.:
```python
    # (universal compile() function - just takes kwargs instead of target-specific options)
    def compile(
        self,
        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
        **kwargs: Any,
    ):
        if self.name().startswith("qscout_"):
            return self.qscout_compile(circuits, **kwargs)
        ...

    def qscout_compile(
        self,
        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
        mirror_swaps: bool = False,
        base_entangling_gate: str = "xx",
        **kwargs,
    ):
        """
        <qscout-specific docs>
        """
        if not self.name().startswith("qscout_"):
            raise ValueError(...)

        if base_entangling_gate not in ("xx", "zz"):
            raise ValueError("base_entangling_gate must be either 'xx' or 'zz'")

        options = {
            "base_entangling_gate": base_entangling_gate,
            "mirror_swaps": mirror_swaps,
            **kwargs
        }
        ...
```
it's maybe a bit redundant, but also nice because it preserves the extended docstrings/references we have in `SuperstaQProvider` for specific users and makes it clear which options apply to which targets, while removing arguments and documentation from `compile()` that don't apply to most devices. It also avoids all the switches in `compile` for specific options, and ensures that e.g. compilation to qscout \*always\* includes values for the `mirror_swaps` and `base_entangling_gates` in the options dict while other targets do not

users would still be able to use either function, because e.g.
```python
backend.compile(circuit, mirror_swaps=False)
```
would do exactly the same thing as
```python
backend.qscout_compile(circuit, mirror_swaps=False),
```
it just wouldn't have all the qscout-specific documentation

Reply:
Great ideas -- done.